packages/terra-clinical-modal-manager/src/components/ModalManager.jsx

```diff
@@ -0,0 +1,198 @@
+import AppDelegate from 'terra-clinical-app-delegate';
+import SlideGroup from 'terra-clinical-slide-group';
+import getBreakpoints from 'terra-responsive-element/lib/breakpoints';
+
+import './ModalManager.scss';
+
+const propTypes = {
+  /**
+   * The AppDelegate instance provided by the containing component. If present, its properties will propagate to the children components.
+   **/
+  app: AppDelegate.propType,
+
+  /**
+   * Components that will receive the ModalManager's AppDelegate configuration. Components given as children must appropriately handle an `app` prop.
+   **/
+  children: PropTypes.node,
+
+  /**
+   * From `connect`. The Array of component keys that will be used to instantiate the Modal's inner components.
+   **/
+  modalComponentKeys: PropTypes.array,
+
+  /**
+   * From `connect`. An Object containing component data used to instantiate the Modal's inner components.
+   **/
+  modalComponentData: PropTypes.object,
+
+  /**
+   * From `connect`. The desired size of the modal.
+   **/
+  size: PropTypes.oneOf(['tiny', 'small', 'medium', 'large', 'huge']),
+
+  /**
+   * From `connect`. The presentation state of the modal.
+   **/
+  isOpen: PropTypes.bool,
+
+  /**
+   * From `connect`. The maximization state of the modal.
+   **/
+  isMaximized: PropTypes.bool,
+
+  /**
+   * From `connect`. A function that dispatches an `open` action.
+   **/
+  openModal: PropTypes.func.isRequired,
+
+  /**
+   * From `connect`. A function that dispatches a `close` action.
+   **/
+  closeModal: PropTypes.func.isRequired,
+
+  /**
+   * From `connect`. A function that dispatches a `push` action.
+   **/
+  pushModal: PropTypes.func.isRequired,
+
+  /**
+   * From `connect`. A function that dispatches a `pop` action.
+   **/
+  popModal: PropTypes.func.isRequired,
+
+  /**
+   * From `connect`. A function that dispatches a `maximize` action.
+   **/
+  maximizeModal: PropTypes.func.isRequired,
+
+  /**
+   * From `connect`. A function that dispatches a `minimize` action.
+   **/
+  minimizeModal: PropTypes.func.isRequired,
+};
+
+const defaultProps = {
+  isOpen: false,
+  isMaximized: false,
+  size: 'small',
+};
+
+class ModalManager extends React.Component {
+  constructor(props) {
+    super(props);
+
+    // I'm tracking the responsive-fullscreen state outside of React and Redux state to limit the number of
+    // renderings that occur.
+    this.forceFullscreenModal = false;
+
+    this.updateFullscreenState = this.updateFullscreenState.bind(this);
+    this.buildModalComponents = this.buildModalComponents.bind(this);
+  }
+
+  componentDidMount() {
+    this.updateFullscreenState();
+    window.addEventListener('resize', this.updateFullscreenState);
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('resize', this.updateFullscreenState);
+  }
+
+  updateFullscreenState() {
+    const previousFullscreenState = this.forceFullscreenModal;
+
+    this.forceFullscreenModal = window.innerWidth < getBreakpoints().small;
```

Comment:
In past implementations, we've forced the modal to be fullscreen when the window size is below small. We want to do this regardless of the actual desired modal size that the consumer specified. We could theoretically have different versions of the 'fullscreen' that flex based on the desired size, but that's not something we've talked about yet.